### PR TITLE
Sync Data mapper code action with latest ballerina api

### PR DESCRIPTION
**Pull request by isuruRuhu**
## Purpose
Sync Data mapper code action with latest ballerina api

## Remarks
- Add unit tests.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java [hunk 1]
```diff
@@ -64,17 +81,150 @@
...
+
+    /**
+     * Get the textDocument/completion response.
+     *
+     * @param filePath Path of the Bal file
+     * @param position Cursor Position
+     * @param endpoint Service Endpoint to Language Server
+     * @return {@link String}   Response as String
+     */
+    public static String getCompletionResponse(String filePath, Position position, Endpoint endpoint) {
```
Comment (rasika) [inline misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java:152]: Shall we remove unnecessary methods from Test Util?

File: misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java [hunk 2]
```diff
@@ -64,17 +81,150 @@
...
 
+    /**
+     * Get the textDocument/hover response.
+     *
+     * @param filePath        Path of the Bal file
+     * @param position        Cursor Position
+     * @param serviceEndpoint Service Endpoint to Language Server
+     * @return {@link String}   Response as String
+     */
+    public static String getHoverResponse(String filePath, Position position, Endpoint serviceEndpoint) {
```
Comment (rasika) [inline misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java:125]: Shall we remove unnecessary methods from Test Util? Please revisit all methods in this class

File: misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java (hunk 1 above)
Comment (isuruRuhu) [reply misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java]: Removed the unnecessary methods

File: misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java (hunk 2 above)
Comment (isuruRuhu) [reply misc/ballerinalang-data-mapper/src/test/java/org/ballerinalang/datamapper/util/TestUtil.java]: Removed unnecessary methods

**Comment by isuruRuhu** [comment]
Checked the data mapper code action disabling capability within VsCode as well as through the plugin. 
Also, manually tested the data mapper code action on VsCode.

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**